### PR TITLE
Updated pull command

### PR DESCRIPTION
**Pull request by pramit-d**
Updated BeSman pull command. https://github.com/Be-Secure/BeSman/issues/103

Command - `$ bes pull --playbook <playbook name> -V <version>`

[peekSZ2KK2.webm](https://github.com/Be-Secure/BeSman/assets/125947785/7121e90d-7a60-4da2-8700-1054b86f859b)

### CODE REVIEW

File: src/main/bash/commands/besman-pull.sh [hunk 1]
```diff
@@ -1,47 +1,75 @@
 #!/bin/bash
 
 function __bes_pull
-{   __besman_check_for_gh || return 1
+{   
+    __besman_check_for_gh || return 1
```
Comment (asa1997) [inline src/main/bash/commands/besman-pull.sh:5]: Dont need this

File: src/main/bash/commands/besman-pull.sh [hunk 2]
```diff
@@ -1,47 +1,75 @@
...
     __besman_check_github_id $BESMAN_USER_NAMESPACE || return 1
     __besman_gh_auth $BESMAN_USER_NAMESPACE
     local type repo dir remote branch return_val namespace
     type=$1
```
Comment (asa1997) [inline src/main/bash/commands/besman-pull.sh:9]: Not required, we are only pulling playbooks

File: src/main/bash/commands/besman-pull.sh [hunk 3]
```diff
@@ -1,47 +1,75 @@
...
+    else
+        __besman_check_url_valid "$lifecyle_file_url" || return 1
+        touch "$lifecycle_file"
+        __besman_secure_curl "$lifecyle_file_url" >>"$lifecycle_file"
+        __besman_echo_green "Playbook $playbook_name $playbook_version added successsfully."
+    fi
+    unset lifecycle_file lifecyle_file_url playbook_name playbook_version
+}
+
+function __besman_check_url_valid() {
```
Comment (asa1997) [inline src/main/bash/commands/besman-pull.sh:54]: The function is already present under path helpers

**Review by asa1997 — CHANGES_REQUESTED** [review]
@pramit-d Please address the comments

File: src/main/bash/commands/besman-pull.sh (hunk 3 above)
Comment (pramit-d) [reply src/main/bash/commands/besman-pull.sh]: @asa1997 removed check url method.

File: src/main/bash/commands/besman-pull.sh (hunk 2 above)
Comment (pramit-d) [reply src/main/bash/commands/besman-pull.sh]: @asa1997  removed __besman_check_for_gh || return 1

File: src/main/bash/commands/besman-pull.sh [hunk 4]
```diff
@@ -1,47 +1,75 @@
 #!/bin/bash
 
 function __bes_pull
-{   __besman_check_for_gh || return 1
+{   
+    __besman_check_for_gh || return 1
     __besman_check_github_id $BESMAN_USER_NAMESPACE || return 1
     __besman_gh_auth $BESMAN_USER_NAMESPACE
```
[the remark this answers is not in the document]
Comment (pramit-d) [reply src/main/bash/commands/besman-pull.sh]: @asa1997  removed __besman_check_for_gh || return 1

File: src/main/bash/commands/besman-pull.sh (hunk 1 above)
Comment (pramit-d) [reply src/main/bash/commands/besman-pull.sh]: @asa1997  removed __besman_check_for_gh || return 1

File: src/main/bash/commands/besman-pull.sh [hunk 5]
```diff
@@ -1,47 +1,49 @@
...
-    fi
-    if [[ $namespace == $BESMAN_NAMESPACE ]]; then
-        remote=upstream
-    else
-        remote=origin
-    fi
-    branch=main
+    playbook_name=$2
+    playbook_version=$3
+    dir=$BESMAN_DIR/playbooks
```
Comment (asa1997) [inline src/main/bash/commands/besman-pull.sh:9]: Please use the env var BESMAN_PLAYBOOK_DIR

Comment (pramit-d) [reply src/main/bash/commands/besman-pull.sh]: @asa1997 updated.

**Comment by asa1997** [comment]
Closing

**Pull request closed without merging**